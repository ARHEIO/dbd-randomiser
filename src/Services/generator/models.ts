export enum IRank {
  BROWN = 1,
  YELLOW,
  GREEN,
  PURPLE,
  IRIDESCANT,
  GOLD
}

export interface IBasic {
  name: string;
  icon: string;
  rank: IRank | null;
}

export interface IModifiable extends IBasic {
  upgradables: IBasic[]
}

export type Killer = IModifiable; // doesn't use rank
export type Item = IModifiable;
export type Survivor = IBasic;
export type Addon = IBasic;
export type Perk = IBasic;
