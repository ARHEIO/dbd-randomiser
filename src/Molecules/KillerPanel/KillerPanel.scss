.killer {
  display: grid;
  grid-gap: 16px;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(4, 1fr);
  margin-bottom: 32px;

  img {
    max-width: 100%;
    max-height: 140px;
  }

  & > * {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-self: center;
    border-bottom: 2px solid #FF2A2B;
    padding-bottom: 16px;
    max-width: 160px;
  }

  &_name {
    grid-column: span 2;
  }

  &_addons,
  &_perks {
    grid-column: span 1;
  }

  @media (min-width: 960px) {
    grid-template-columns: repeat(4, 1fr);
    grid-template-rows: repeat(2, 1fr);
  }
}
