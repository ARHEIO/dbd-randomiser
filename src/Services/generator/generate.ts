import axios from 'axios';
import {
  GeneratedKiller, GeneratedSurvivor,
} from './models';
import config from '../../config';

export const generateKiller = (): Promise<{data: GeneratedKiller}> => axios.get(`${config.endpoints.randomiserApi}/killer`);

export const generateSurvivor = (): Promise<{data: GeneratedSurvivor}> => axios.get(`${config.endpoints.randomiserApi}/survivor`);
