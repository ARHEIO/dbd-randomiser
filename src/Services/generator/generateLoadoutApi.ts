import axios from 'axios';
import { GeneratedKiller, GeneratedSurvivor } from './models';
import config from '../../config';

export const generateKiller = (): Promise<{data: GeneratedKiller}> => axios.get(`${config.endpoints.randomiserApi}/loadout?q=killer`);

export const generateSurvivor = (): Promise<{data: GeneratedSurvivor}> => axios.get(`${config.endpoints.randomiserApi}/loadout?q=survivor`);
