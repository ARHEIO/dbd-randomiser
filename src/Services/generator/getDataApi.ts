import axios from 'axios';
import config from '../../config';
import {
  Killer, Survivor, Item, Perk,
} from './models';

export const getKiller = (id: number): Promise<{data: Killer}> => axios
  .get(`${config.endpoints.randomiserApi}/killer/${id}`);
export const getKillerWithAddons = (id: number): Promise<{data: Killer}> => axios
  .get(`${config.endpoints.randomiserApi}/killer/${id}/?expands=upgradables`);
export const getKillers = (): Promise<{data: Killer[]}> => axios
  .get(`${config.endpoints.randomiserApi}/killer`);

export const getSurvivor = (id: number): Promise<{data: Survivor}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor/${id}`);
export const getSurvivors = (): Promise<{data: Survivor[]}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor`);

export const getItem = (id: number): Promise<{data: Item}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor/${id}`);
export const getItemWithAddons = (id: number): Promise<{data: Item}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor/${id}?expands=upgradables`);
export const getItems = (): Promise<{data: Item[]}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor`);

export const getSurvivorPerk = (id: number): Promise<{data: Perk[]}> => axios
  .get(`${config.endpoints.randomiserApi}/perk/survivor/${id}`);
export const getSurvivorPerks = (): Promise<{data: Perk}> => axios
  .get(`${config.endpoints.randomiserApi}/perk/survivor`);

export const getKillerPerk = (id: number): Promise<{data: Perk}> => axios
  .get(`${config.endpoints.randomiserApi}/perk/killer/${id}`);
export const getKillerPerks = (): Promise<{data: Perk[]}> => axios
  .get(`${config.endpoints.randomiserApi}/perk/killer`);
