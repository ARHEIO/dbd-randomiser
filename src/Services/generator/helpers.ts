export const randomNumberGenerator = (max: number) => {
  return Math.floor(Math.random() * max)
}

export const getMultiple = (arrayOfThings: any[], numberToGet: number) => {
  return arrayOfThings && arrayOfThings.length
    ? arrayOfThings
        .map(x => ({ x, r: Math.random() }))
        .sort((a, b) => a.r - b.r)
        .map(a => a.x)
        .slice(0, numberToGet)
    : [];
}

export const getAddons = (addons: any[]) => getMultiple(addons, 2)

export const getPerks = (perks: any[]) => getMultiple(perks, 4)