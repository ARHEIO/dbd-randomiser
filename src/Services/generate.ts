import { killers, IKiller } from './Killers/killers';
import { killerAddons } from './Addons/addons.killer';
import { killerPerks } from './Perks/perks.killer';
import { randomNumberGenerator, getAddons, getPerks } from './helpers';
import { survivorItems } from './Items/items.survivor';
import { survivorAddons } from './Addons/addons.survivor';
import { SurvivorPerks } from './Perks/perks.survivor';
import { IGameObject } from './Common/game-object.model';

export const generateKiller = () => {
  const killer: IKiller = killers[randomNumberGenerator(killers.length)];
  const selectedAddons = getAddons((<any>killerAddons)[`${killer.link}`]);
  const selectedPerks = getPerks(killerPerks);
  return {
    name: killer.name,
    addons: selectedAddons,
    perks: selectedPerks
  }
}

export const generateSurvivor = () => {
  const item: IGameObject = survivorItems[randomNumberGenerator(survivorItems.length)];
  const selectedAddons = getAddons((<any>survivorAddons)[`${item.link}`]);
  const selectedPerks = getPerks(SurvivorPerks);
  return {
    item: item,
    addons: selectedAddons,
    perks: selectedPerks
  }
}
