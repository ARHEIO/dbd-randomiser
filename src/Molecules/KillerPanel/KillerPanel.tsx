import React, { ReactElement } from 'react';
import './KillerPanel.scss';

import DiamondIcon from '../../Atoms/DiamondIcon/DiamondIcon';
import SquareIcon from '../../Atoms/SquareIcon/SquareIcon';

import { Perk as IPerk } from '../../Services/generator/models';
import { GeneratedKiller } from '../../Services/generator/generate';

const KillerPanel = (props: { killer: GeneratedKiller }): ReactElement => {
  const {
    killer,
  } = props;
  return (
    <div className="killer">
      <SquareIcon name={killer.name} icon={killer.icon} rank={null} />
      {killer.addons.map((addon: any) => (
        <SquareIcon key={addon.name} name={addon.name} icon={addon.icon} rank={addon.rank} />
      ))}
      {killer.perks.map((perk: IPerk) => (
        <DiamondIcon key={perk.name} name={perk.name} icon={perk.icon} rank={perk.rank} />
      ))}
    </div>
  );
};

export default KillerPanel;
