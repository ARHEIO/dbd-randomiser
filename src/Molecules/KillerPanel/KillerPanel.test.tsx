import React from 'react';

import { render } from '@testing-library/react';
import pretty from 'pretty';
import KillerPanel from './KillerPanel';
import * as generators from '../../Services/generator/generate';

const defaultObject: generators.GeneratedKiller = {
  name: 'Adam',
  icon: 'assets/adam.png',
  addons: [
    { icon: '', name: 'Evoluent Vertical Mouse 4', rank: 5 },
    { icon: '', name: 'OLKB Preonic', rank: 5 },
  ],
  perks: [
    { icon: '', rank: 5, name: 'Archetecture' },
    { icon: '', rank: 5, name: 'Dockerised' },
    { icon: '', rank: 5, name: 'Carpel Tunneling' },
    { icon: '', rank: 5, name: '8 Hour Nap' },
  ],
};

describe('Search View', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render when the default object is passed', () => {
    const { container } = render(<KillerPanel killer={defaultObject} />);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
