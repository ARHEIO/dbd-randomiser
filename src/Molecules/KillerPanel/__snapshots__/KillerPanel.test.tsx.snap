// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search View should render when the default object is passed 1`] = `
"<div class=\\"killer\\">
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">Adam</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(97, 81, 60);\\"><img alt=\\"Adam\\" src=\\"https://assets.arhe.io/dbd-randomiser/shitty/assets/adam.png\\"></div>
  </div>
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">Evoluent Vertical Mouse 4</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Evoluent Vertical Mouse 4\\" src=\\"https://assets.arhe.io/dbd-randomiser/shitty/\\"></div>
  </div>
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">OLKB Preonic</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"OLKB Preonic\\" src=\\"https://assets.arhe.io/dbd-randomiser/shitty/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Archetecture</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Archetecture\\" src=\\"https://assets.arhe.io/dbd-randomiser/shitty/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Dockerised</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Dockerised\\" src=\\"https://assets.arhe.io/dbd-randomiser/shitty/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Carpel Tunneling</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Carpel Tunneling\\" src=\\"https://assets.arhe.io/dbd-randomiser/shitty/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">8 Hour Nap</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"8 Hour Nap\\" src=\\"https://assets.arhe.io/dbd-randomiser/shitty/\\"></div>
  </div>
</div>"
`;
